### DISCUSSION TITLE
Fallback page is not displayed until getStaticProps is done

### DISCUSSION THREAD

**Discussion by gkielwasser** [vercel/next.js#44965] [Help]
### Verify canary release

- [X] I verified that the issue exists in Next.js canary release

### Provide environment information

 Operating System:
      Platform: win32
      Arch: x64
      Version: Windows 10 Pro for Workstations
    Binaries:
      Node: 14.18.3
      npm: N/A
      Yarn: N/A
      pnpm: N/A
    Relevant packages:
      next: 12.1.4
      react: 17.0.2
      react-dom: 17.0.2

### What browser are you using? (if relevant)

Chrome -  99.0.4844.84

### How are you deploying your application? (if relevant)

Bug is detected on local and on Vercel

### Describe the Bug

The fallback page is not displayed while getStaticProps is running.

The fallback page is well displayed if the page is directly accessed without routing involved (aka: open directly the page in a new tab).

### Expected Behavior

Fallback page should be displayed instantly even if getStaticProps is running.

### To Reproduce

Add page "/isr-src.tsx"
```
import React from 'react'
import Link from "Next/link";


export async function getStaticProps() {
  return {
    props: {}
  }
}

const Page = function () {
  return <div>
    Vercel test ISR - source page
    <div>
      {
        Array.from(Array(50).keys()).map(i => <Link href={`/isr-target/${i}`} prefetch={false} key={i}>page {i}</Link>)
      }
    </div>

  </div>
}
export default Page
````

Add page  "/isr-target/[slug].tsx"
```
import { GetStaticPropsContext } from "next";
import { useRouter } from "next/router";
import Link from "next/Link";
import React from "react";

export async function getStaticPaths() {
  return {
    paths: [],
    fallback: true
  }
}

export async function getStaticProps(context: GetStaticPropsContext) {
  await new Promise(r => setTimeout(r, 4000));

  return {
    props: {},
    revalidate: 10
  }
}

const Page = function () {
  const { isFallback } = useRouter()
  return <div>
    Vercel test ISR - target page
    <br />
    {isFallback ? 'Loading' : 'Final page'}
    <br/>
    <Link href={'/isr-src'} prefetch={false}><a>back</a></Link>
  </div>

}
export default Page

```

To reproduce:
- start from page "/isr-src"
- open network tab
- because I have used a next/link the hover will start the prefetch. Please care to direct click the link you want to test.
- click one link
- during the fetching of the page we can clairly see in network, the fallback page is not displayed

**Comment by ijjk** [comment]
Hi, the fallback page is only expected to show when visiting the page directly and not during client-transitions with `next/link`.

**Comment by maurodesouza** [comment]
@ijjk facing the same issue here even when i'm on the same fallback page, or i'm doing something very wrong 😅

My dynamic route
https://github.com/maurodesouza/star-wars-explorer/blob/master/src/pages/universe/%5B%5B...slug%5D%5D.tsx

My page template
https://github.com/maurodesouza/star-wars-explorer/blob/master/src/templates/Entity/index.tsx

**Comment by gkielwasser** [comment]
> Hi, the fallback page is only expected to show when visiting the page directly and not during client-transitions with `next/link`.

Because it involves only Nextjs utilities (ISR, next/link), what is expected to happen for such a case ?

Does it means man should not use `next/link` if it targets pages that would involve validation/revalidation ? It seems tricky, it means client routing logic should "know" about page logic revalidation strategy. In other terms, targeting any pages using `revalidate` should not be done using `next/link` ?

According to what you say, it seems to be wrong use of `next/link` my side, I'm using almost everywhere to performs internal transitions. My pages have different revalidation strategy applied. The result is I have unpredictable delay while navigating into the app.

Thanks for your feeback

**Comment by carlosrsabreu** [comment]
> Hi, the fallback page is only expected to show when visiting the page directly and not during client transitions with `next/link`.

How do users know that page is loading if the fallback page appears when visiting the page directly? How do they get feedback that the requested page is loading when visited using `next/link` through other project pages? I'm facing the same problem as described with more details [here](https://github.com/vercel/next.js/discussions/23992#discussioncomment-2602416).

Thank you!

**Comment by carlosfaria94** [comment]
> Hi, the fallback page is only expected to show when visiting the page directly and not during client-transitions with `next/link`.

IMO it doesn't make sense. The fallback page should appear on client-transitions with `next/link`. I think it's an expected behavior...

**Comment by unispacesio** [comment]
what's the point of router.isFallback then? It a bad UX to show nothing happening until the page is loaded. Please fix this

**Comment by franco14lorenzo** [comment]
I'm here because I have the same problem. Is this issue being worked on? I think this fix could improve the UX considerably.
